### PR TITLE
[terra-menu] Updated docs to use MDX syntax

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Updated terra-menu docs to use MDX syntax

### CODE REVIEW

File: packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+[View Component Source Code](https://github.com/cerner/terra-framework/tree/master/packages/terra-menu)
+
+## Examples 
+<BasicMenu title='Default Menu' />
+<ArrowMenu title='Menu With Arrow' />
+<MenuBounded title='Menu Bounded' />
+<MenuWidths title='Menu Widths' />
+<MenuWithInstructionsForUse title='Help Menu Item: Electronic Instructions For Use' description="The Electronic Instructions For Use (eIFU) icon is a regulatory requirement for CE Mark Certification and Compliance, and is used to indicate Help content that is the equivalent of a manufacturer\'s instruction manual. The label for the Help menu option should be the name of the application followed by 'Help.'"  />
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: Can you break this into new lines for readability?

File: packages/terra-menu/src/terra-dev-site/doc/example/ArrowMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import Button from 'terra-button';
+import PropTypes from 'prop-types';
+import Menu from 'terra-menu';
+import classNames from 'classnames/bind';
+import styles from './BasicMenu.module.scss';
+
+const cx = classNames.bind(styles);
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/terra-dev-site/doc/example/ArrowMenu.jsx]: Can you explain why this was added? Is this replacing the usage of 
`<BasicMenu isArrowDisplayed />` seen [here](https://github.com/cerner/terra-framework/pull/1056/files#diff-f54a0ae78bc54090261bdc8f031abe0bL36)?

File: packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+|-|-|
+| react | ^16.8.5 |
+| react-dom | ^16.8.5 |
+
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+## Usage
+
+```jsx
+import Menu from 'terra-menu';
```
Comment (emilyrohrbough) [inline packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: How would I use Menu Item and Menu Item Group?

File: packages/terra-menu/src/terra-dev-site/doc/example/ArrowMenu.jsx (hunk 2 above)
Comment (saket2403) [reply packages/terra-menu/src/terra-dev-site/doc/example/ArrowMenu.jsx]: Passing `isArrowDisplayed` as prop in mdx file was not working as expected so, I made a new example file for it.

File: packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import MenuPropsTable from 'terra-menu/src/Menu?dev-site-props-table';
+import MenuItemPropsTable from 'terra-menu/src/MenuItem?dev-site-props-table';
+import MenuItemGroupPropsTable from 'terra-menu/src/MenuItemGroup?dev-site-props-table';
+
+import ArrowMenu from '../example/ArrowMenu?dev-site-example';
+import BasicMenu from '../example/BasicMenu?dev-site-example';
+import MenuBounded from '../example/MenuBounded?dev-site-example';
+import MenuWidths from '../example/MenuWidths?dev-site-example';
+import MenuWithInstructionsForUse from '../example/MenuWithInstructionsForUse?dev-site-example';
+
```
Comment (avinashg1994) [inline packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: Can you add the "Badge" here

Comment (saket2403) [reply packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: Added 7dfc345

File: packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx (hunk 3 above)
Comment (saket2403) [reply packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: For Menu Item and Menu Item group separate imports are not required `Menu.Item` and `Menu.ItemGroup` works. So, added a description 7dfc345

File: packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx (hunk 1 above)
Comment (saket2403) [reply packages/terra-menu/src/terra-dev-site/doc/menu/Menu.1.doc.mdx]: Fixed 7dfc345

**Pull request merged**